### ISSUE TITLE
[4.0] Date picker is not accessible for keyboard user

### ISSUE THREAD

**Issue by zwiastunsw** [joomla/joomla-cms#24019]
### Steps to reproduce the issue
1. Go to Joomla Backend
2. Go to Content/Article/New or Edit / Publishing tab 
  - Use Tab key to move focus to calendar field 
   - Use Down arrow to open calendar dialog (not work, use tab to move fokus to calendar icon and use Spacebar to open calendar dialog)
   - Use Down arrow to move focus to current date
   - Use Left/Right/Down/Up arrows to select date
   - Use Tab key or Shift + Tab key to move focus to Clear/Today/Close buttons or to select month/year (not works)
   - Use Space/Enter key to place selected date into field
### Expected result
You should perform all of the following operations 
### Actual result
You can only select a date. You cannot select the month, year, time selection options, buttons Clear/Today/Close.   

### Additional comments
See: [Date Picker Example Using ARIA 1.1 Combobox](https://raw.githack.com/w3c/aria-practices/issue34-date-picker-examples/examples/datepicker/datepicker.html) as an example of accessible date picker

**Comment by dgrammatiko** [comment]
I think @C-Lodder is working on this

**Comment by hardik-codes** [comment]
May I assist you in this @C-Lodder if you are working on it ?

**Comment by C-Lodder** [comment]
When get some time again I will start on it again after a first failed attempt. The code isn't documented at all so I'm in 2 minds whether or not to try again or extend something that's already out there

**Comment by C-Lodder** [comment]
@hardik-codes - If you're comfortable with JS and familiar with Custom Elements, feel free to take it on. It'll be fairly big task

**Comment by brianteeman** [comment]
There are several examples of fully accessible datepickers here https://www.webaxe.org/accessible-date-pickers/

**Comment by brianteeman** [comment]
and there is an even better solution which is zero maintenance and fully accessible - however it only supports gregorian calendars. I assume @infograf768 that this would not be acceptable unless there was some way for it to fall back to a scripted version for non-gregorian calendars?

**Comment by dgrammatiko** [comment]
@brianteeman AFAIK Joomla using a date-time picker not a date-picker so your suggestions are not fit

**Comment by brianteeman** [comment]
The only solution I can find is to separate the date and time

**Comment by zwiastunsw** [comment]
Unfortunately, I also don't know a good solution for the date and time selector in one component. Creating a separate date and time selector seems to be a good idea. 
HTML5 offers simple html controls for date and time selection (input type="date" and input type="time").  These have limitations, but are accessible and well supported by browsers.
It can also be assumed that if the user can enter the date and time in the text field, the accessibility requirements are met. However, you need to make sure that you have accurate information about the date and time format. 

[Maybe this will help](https://ej2.syncfusion.com/documentation/datetimepicker/accessibility/)

**Comment by C-Lodder** [comment]
@zwiastunsw Shame FF and Safari don't support this: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local

**Comment by zwiastunsw** [comment]
@C-Lodder : I know Safari does not yet support these controls as expected, but it will display text fields. So accessibility will also be provided.

**Comment by brianteeman** [comment]
As far as I am aware the native html date and datetime pickers do not support non-gregorian calendars

**Comment by zwiastunsw** [comment]
I don't know that. I'm sorry.

**Comment by coolcat-creations** [comment]
This is still an issue in J5
